### PR TITLE
[release/0.28] Revert commit 88737d AKA PR #3903.

### PR DESCRIPTION
**Pull request by ksbrar**
Revert https://github.com/microsoft/FluidFramework/pull/3903 since ODC does not have necessary feature for it yet and (and they might start integrating 0.28 before they do).

### CODE REVIEW

**Comment by jatgarg** [comment]
> ⯆ @fluidframework/base-host: -4 Bytes
> ⯅ @fluid-example/bundle-size-tests: +61.82 KB
> Metric Name	Baseline Size	Compare Size	Size Diff
> container.js	180.09 KB	180.13 KB	⯅ +38 Bytes
> map.js	39.79 KB	39.79 KB	■ No change
> matrix.js	137.45 KB	137.45 KB	■ No change
> odspDriver.js	191.48 KB	253.23 KB	⯅ +61.75 KB
> sharedString.js	167.6 KB	167.64 KB	⯅ +38 Bytes
> Total Size	716.41 KB	778.23 KB	⯅ +61.82 KB
> Baseline commit: [d746f91](https://github.com/microsoft/FluidFramework/commit/d746f912c2b080bc0f02af8ca73144ba6894194c)
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [7ee079d](https://github.com/microsoft/FluidFramework/commit/7ee079dbc24ad67f76e99f51c750bf91f8509d56)

@heliocliu @ksbrar Why is this adding 60kb to the driver?

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 1]
```diff
@@ -102,6 +103,10 @@ function convertSummaryIntoContainerSnapshot(createNewSummary: ISummaryTree) {
         throw new Error("App and protocol summary required for create new path!!");
     }
     const documentAttributes = getDocAttributesFromProtocolSummary(protocolSummary);
+    // Currently for the scenarios we have we don't have ops in the detached container. So the
+    // sequence number would always be 0 here. However odsp requires to have at least 1 snapshot.
+    assert(documentAttributes.sequenceNumber === 0, "Sequence No for detached container snapshot should be 0");
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/createFile.ts:108]: I think @jatgarg removed this assert in later iterations, so likely that's incorrect merge (or missing change?)

**Review by vladsud — APPROVED** [review]
Thanks!

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:108]: Yes, I modified this asseret. Could you please change it to that?

**Comment by heliocliu** [comment]
> 
> 
> > ⯆ @fluidframework/base-host: -4 Bytes
> > ⯅ @fluid-example/bundle-size-tests: +61.82 KB
> > Metric Name	Baseline Size	Compare Size	Size Diff
> > container.js	180.09 KB	180.13 KB	⯅ +38 Bytes
> > map.js	39.79 KB	39.79 KB	■ No change
> > matrix.js	137.45 KB	137.45 KB	■ No change
> > odspDriver.js	191.48 KB	253.23 KB	⯅ +61.75 KB
> > sharedString.js	167.6 KB	167.64 KB	⯅ +38 Bytes
> > Total Size	716.41 KB	778.23 KB	⯅ +61.82 KB
> > Baseline commit: [d746f91](https://github.com/microsoft/FluidFramework/commit/d746f912c2b080bc0f02af8ca73144ba6894194c)
> > Generated by 🚫 [dangerJS](https://danger.systems/js) against [7ee079d](https://github.com/microsoft/FluidFramework/commit/7ee079dbc24ad67f76e99f51c750bf91f8509d56)
> 
> @heliocliu @ksbrar Why is this adding 60kb to the driver?

@christiango you almost got your first victim 👀  Christian removed usages of `assert` in https://github.com/microsoft/FluidFramework/pull/4144.  You need to remove the assert call that got added back with the implementation from common-utils

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 2]
```diff
@@ -3,11 +3,12 @@
  * Licensed under the MIT License.
  */
 
+import { strict as assert } from "assert";
```
Comment (heliocliu) [inline packages/drivers/odsp-driver/src/createFile.ts:6]: use `import { assert } from "@fluidframework/common-utils";` instead

**Review by heliocliu — CHANGES_REQUESTED** [review]
need to use common-utils assert implementation to avoid bundle size increase.  that change just went in yesterday, after the change you're reverting i'd guess

**Comment by christiango** [comment]
> > > ⯆ @fluidframework/base-host: -4 Bytes
> > > ⯅ @fluid-example/bundle-size-tests: +61.82 KB
> > > Metric Name	Baseline Size	Compare Size	Size Diff
> > > container.js	180.09 KB	180.13 KB	⯅ +38 Bytes
> > > map.js	39.79 KB	39.79 KB	■ No change
> > > matrix.js	137.45 KB	137.45 KB	■ No change
> > > odspDriver.js	191.48 KB	253.23 KB	⯅ +61.75 KB
> > > sharedString.js	167.6 KB	167.64 KB	⯅ +38 Bytes
> > > Total Size	716.41 KB	778.23 KB	⯅ +61.82 KB
> > > Baseline commit: [d746f91](https://github.com/microsoft/FluidFramework/commit/d746f912c2b080bc0f02af8ca73144ba6894194c)
> > > Generated by 🚫 [dangerJS](https://danger.systems/js) against [7ee079d](https://github.com/microsoft/FluidFramework/commit/7ee079dbc24ad67f76e99f51c750bf91f8509d56)
> > 
> > 
> > @heliocliu @ksbrar Why is this adding 60kb to the driver?
> 
> @christiango you almost got your first victim 👀 Christian removed usages of `assert` in #4144. You need to remove the assert call that got added back with the implementation from common-utils

Hmm, the build didn't fail. Not sure if the build here kicked off after I merged the webpack plugin. Is that why we didn't catch it?

**Comment by heliocliu** [comment]
@christiango this is targeting the 0.28 branch which is why

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 2 above)
Comment (ksbrar) [reply packages/drivers/odsp-driver/src/createFile.ts:6]: woops!

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 1 above)
Comment (heliocliu) [reply packages/drivers/odsp-driver/src/createFile.ts:108]: Looks like this actually would cover the assert issue as well oops

**Review by heliocliu — APPROVED** [review]

**Review by jatgarg — APPROVED** [review]

**Pull request merged**